src/number.c

```diff
@@ -384,7 +384,9 @@ static void print_complex(SCM n, SCM port, int mode)
   SCM imag = COMPLEX_IMAG(n);
 
   STk_print(COMPLEX_REAL(n), port, mode);
-  if (positivep(imag) || (REALP(imag) && !signbit(REAL_VAL(imag))))
```

Comment:
I read and re-read this code and don't understand why it works also for  NaNs (but obviously it works).  There is no call to `isnan`. Why does it works?

Reply:
Hi @egallesio !
I suppose it's because NaNs and infinities do have a signbit and they satisfy `REALP`, so

* `+nan.0` passes the first line in the condition (sign bit is zero). But for the extra plus to be printed, the other two lines require:
- Either zero (it's not), OR:
- Positive AND not infinity -- but NaNs so not satisfy the STklos C predicate `positivep`. :)

* `-nan.0` fails the first line, because its sign bit is 1 (negated = 0). Then the `+` is not printed, and the `-nan.0` shows up.

I'll include a comment in the code, I think it's kind of obscure...

Reply:
(The sign bit for NaNs and infinities is described in the manpage for `signbit`, and as far as I know it's standard)